multiDict.d.ts

```diff
@@ -0,0 +1,105 @@
+type GenericConstructor = new (...args: unknown[]) => object
+
+interface TreeOptions<
+	DType extends GenericConstructor,
+	Types extends GenericConstructor[]
+> {
+	defaultType?: DType
+	types?: Types
+}
+
+declare class Tree<
+	Keys extends any[],
+	Val,
+	DType extends GenericConstructor = typeof Map,
+	Types extends GenericConstructor[] = []
+> {
+	private root: DType
+	public options: TreeOptions<DType, Types>
+	constructor(options: TreeOptions<DType, Types>)
+
+	get(keys: Keys, value: false): DType | Types[number] | undefined
+	get(keys: Keys, value: true): Val | undefined
+	set(keys: Keys, value: Val): void
+	has(keys: Keys): boolean
+	clear(): void
+	level(keys: Keys): Object[]
+	private createNode(depth: number): Object
```

Comment:
I think the return type should be `DType | Types[number]` because `get` also has this.